sdk/tests/conformance/extensions/ext-polygon-offset-clamp.html

```diff
@@ -0,0 +1,172 @@
+    debug("");
+    debug("Check state update using the new function");
+    ext.polygonOffsetClampEXT(1.0, 2.0, 3.0);
+    shouldBe('gl.getParameter(gl.POLYGON_OFFSET_FACTOR)', '1.0');
+    shouldBe('gl.getParameter(gl.POLYGON_OFFSET_UNITS)', '2.0');
+    shouldBe('gl.getParameter(ext.POLYGON_OFFSET_CLAMP_EXT)', '3.0');
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
+    debug("");
+    debug("Check that the unextended function resets the clamp value to zero");
+    gl.polygonOffset(4.0, 5.0);
+    shouldBe('gl.getParameter(gl.POLYGON_OFFSET_FACTOR)', '4.0');
+    shouldBe('gl.getParameter(gl.POLYGON_OFFSET_UNITS)', '5.0');
+    shouldBe('gl.getParameter(ext.POLYGON_OFFSET_CLAMP_EXT)', '0.0');
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
+}
+
+function checkClamping() {
+    debug("");
+    debug("Check polygon offset clamp operation");
+
+    // The shader creates a depth slope from left (0) to right (1).
+    //
+    // This test issues two draw calls:
+    //
+    //           Draw 2 (green): factor width, offset 0, clamp 0.5, depth test: Greater
+    //          ^     |       __________________
+    //          |     |    __/      __/
+    //          |     V __/      __/  <--- Draw 1 (red): factor width, offset 0, clamp 0.25
+    //          |    __/      __/
+    //          | __/      __/
+    //          |/      __/
+    //          |    __/
+    //          | __/
+    //          |/
+    //          |
+    //          |
+    //          +------------------------------->
+    //
+    // Result:  <---------green--------><--red-->
+
+
+    const program = wtu.setupProgram(gl, [wtu.simpleVertexShader,
+                                          wtu.simpleColorFragmentShader]);
+    gl.useProgram(program);
+    const colorLoc = gl.getUniformLocation(program, "u_color");
+
+    const buf = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, buf);
+    gl.bufferData(
+        gl.ARRAY_BUFFER,
+        new Float32Array([-1, -1, -1, 1, -1, 1, -1, 1, -1, 1, 1, 1]),
+        gl.STATIC_DRAW);
+    gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 0, 0);
+    gl.enableVertexAttribArray(0);
+
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
+
+    gl.enable(gl.POLYGON_OFFSET_FILL);
+    gl.clearColor(0, 0, 0, 1);
+    gl.clearDepth(0);
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+    gl.enable(gl.DEPTH_TEST);
+
+    gl.depthFunc(gl.ALWAYS);
+    gl.uniform4f(colorLoc, 1, 0, 0, 1);
+    ext.polygonOffsetClampEXT(w, 0, 0.25);
+    gl.drawArrays(gl.TRIANGLE_STRIP, 0, 4);
+
+    gl.depthFunc(gl.GREATER);
+    gl.uniform4f(colorLoc, 0, 1, 0, 1);
+    ext.polygonOffsetClampEXT(w, 0, 0.5);
+    gl.drawArrays(gl.TRIANGLE_STRIP, 0, 4);
+
+    wtu.checkCanvasRect(
+        gl,
+        0, 0, 3 * w / 4 - 1, h,
+        [0, 255, 0, 255], "should be green");
+
+    wtu.checkCanvasRect(
+        gl,
+        3 * w / 4 + 1, 0, w / 4 - 1, h,
```

Comment:
Do you think all implementations are likely accurate enough to have this cutover region within a +/-1 pixel region in the rendered canvas?

Reply:
Yes. A similar test passes on ANGLE CI.